File: ulyp-ui/src/main/java/com/ulyp/ui/util/CssClass.java
package com.ulyp.ui.util;

import java.util.Arrays;
import java.util.List;

public enum CssClass {

    CALL_TREE_TYPE_NAME("ulyp-ctt", "ulyp-ctt-typename"),
    CALL_TREE_STRING_LITERAL("ulyp-ctt", "ulyp-ctt-string"),
    CALL_TREE_NUMBER("ulyp-ctt", "ulyp-ctt-number"),
    CALL_TREE_PLAIN_TEXT("ulyp-ctt", "ulyp-ctt-sep"),
    CALL_TREE_COLLECTION_BRACE("ulyp-ctt", "ulyp-ctt-brace"),
    CALL_TREE_IDENTITY_REPR("ulyp-ctt", "ulyp-ctt-identity"),
    CALL_TREE_METHOD_NAME("ulyp-ctt", "ulyp-ctt-method-name"),
    CALL_TREE_RETURN_VALUE("ulyp-ctt", "ulyp-ctt-return-value"),
    CALL_TREE_THROWN("ulyp-ctt", "ulyp-ctt-thrown"),
    CALL_TREE_ARG_NAME("ulyp-ctt", "ulyp-ctt-arg-name"),
    CALL_TREE_CALLEE("ulyp-ctt", "ulyp-ctt-callee");

    private final List<String> classes;

    CssClass(String... classes) {
        this.classes = Arrays.asList(classes);
    }

    public List<String> getCssClasses() {
        return classes;
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/util/CssClass.kt
package com.ulyp.ui.util

import java.util.*

enum class CssClass(vararg classes: String) {

    ERROR_TEXT_CSS("ulyp-error-text"),
    HELP_TEXT_CSS("ulyp-help-text"),
    CALL_TREE_ALL_CSS("ulyp-ctt"),
    CALL_TREE_TYPE_NAME_CSS("ulyp-ctt-typename"),
    CALL_TREE_STRING_LITERAL_CSS("ulyp-ctt-string"),
    CALL_TREE_NUMBER_CSS("ulyp-ctt-number"),
    CALL_TREE_FILE_CSS("ulyp-ctt-file"),
    CALL_TREE_NULL_CSS("ulyp-ctt-null"),
    CALL_TREE_NODE_SEPARATOR_CSS("ulyp-ctt-sep"),
    CALL_TREE_COLLECTION_BRACKET_CSS("ulyp-ctt-bracket"),
    CALL_TREE_IDENTITY_CSS("ulyp-ctt-identity"),
    CALL_TREE_IDENTITY_HASH_CODE_CSS("ulyp-ctt-identity-hash-code"),
    CALL_TREE_METHOD_NAME_CSS("ulyp-ctt-method-name"),
    CALL_TREE_RETURN_VALUE_CSS("ulyp-ctt-return-value"),
    CALL_TREE_THROWN_CSS("ulyp-ctt-thrown"),
    CALL_TREE_ARG_NAME_CSS("ulyp-ctt-arg-name"),
    CALL_TREE_CALLEE_CSS("ulyp-ctt-callee");

    val cssClasses: List<String>
        get() = field

    init {
        cssClasses = Arrays.asList(*classes)
    }
}